### PR TITLE
block on all async operation to completion

### PR DESCRIPTION
**Pull request by anekkanti**
## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: app/request.go [hunk 1]
```diff
@@ -2,11 +2,34 @@ package app
...
 
+const (
+	AsyncRequestFlagName   = "async"
+	RequestTimeoutFlagName = "request-timeout"
+)
+
+var (
+	RequestTimeoutFlag = &cli.DurationFlag{
+		Name:    RequestTimeoutFlagName,
+		Usage:   "Time to wair for asynchronous requests to finish",
```
Comment (mastermanu) [inline app/request.go:22]: [Nit] typo on line 22

File: app/request.go [hunk 2]
```diff
@@ -63,8 +86,90 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+			Action: func(ctx *cli.Context) error {
+				return c.waitOnRequest(ctx, "", ctx.String("request-id"))
 			},
 		}},
 	}}, nil
 }
+
+func (c *RequestClient) waitOnRequest(ctx *cli.Context, operation string, requestID string) error {
+
+	ticker := time.NewTicker(time.Millisecond)
```
Comment (mastermanu) [inline app/request.go:117]: this needs to be less frequent than a millisecond

Comment (mastermanu) [reply app/request.go:117]: ok, I see this is just for the first iteration

File: app/request.go [hunk 3]
```diff
@@ -63,8 +86,90 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+				fmt.Fprintf(writer, "waiting for %s operation to finish, current state: %s\n",
+					operation, request.RequestStatus_State_name[int32(status.State)])
+			} else {
+				fmt.Fprintf(writer, "waiting for request with '%s' id to finish, current state: %s\n",
+					requestID, request.RequestStatus_State_name[int32(status.State)])
+			}
+			ticker.Reset(time.Second * time.Duration(status.CheckDuration.Seconds))
+		}
+	}
+	if operation != "" {
```
Comment (mastermanu) [inline app/request.go:158]: print the requestID as well

File: app/request.go [hunk 4]
```diff
@@ -63,8 +86,90 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+}
+
+func (c *RequestClient) HandleRequestStatus(
+	ctx *cli.Context,
+	operation string,
+	status *request.RequestStatus,
+) error {
+	if ctx.Bool(AsyncRequestFlagName) {
+		return PrintProto(status)
+	}
```
Comment (mastermanu) [inline app/request.go:173]: please print out the request object so if someone ctrl-Cs and wants to check after the fact, they are able to do so

File: app/request.go [hunk 5]
```diff
@@ -2,11 +2,34 @@ package app
...
 	"github.com/urfave/cli/v2"
 )
 
+const (
+	AsyncRequestFlagName   = "async"
+	RequestTimeoutFlagName = "request-timeout"
+)
+
+var (
+	RequestTimeoutFlag = &cli.DurationFlag{
```
Comment (mastermanu) [inline app/request.go:20]: add short-hand flags? (--rt/-t for request-timeout, -a for async)

File: app/request.go [hunk 6]
```diff
@@ -63,8 +86,90 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+				if err != nil {
+					return err
+				}
+				return PrintProto(res)
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait till the request completes",
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
```
Comment (mastermanu) [inline app/request.go:99]: do we need to pass inrequest timeout flag, or is that global?

**Review by mastermanu — APPROVED** [review]

File: app/request.go [hunk 7]
```diff
@@ -63,8 +88,93 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
-				return c.getRequestStatus(ctx.String(NamespaceFlagName), ctx.String("request-id"))
+				res, err := c.getRequestStatus(ctx, ctx.String("request-id"))
+				if err != nil {
+					return err
+				}
+				return PrintProto(res)
+			},
+		}, {
+			Name:    "wait",
+			Usage:   "wait till the request completes",
```
Comment (feihuang) [inline app/request.go:99]: nit: s/wait/Wait

File: app/request.go [hunk 8]
```diff
@@ -63,8 +88,93 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+			Aliases: []string{"w"},
+			Flags: []cli.Flag{
+				NamespaceFlag,
+				&cli.StringFlag{
+					Name:     "request-id",
+					Usage:    "the request-id of the asynchronous request",
```
Comment (feihuang) [inline app/request.go:105]: nit: s/the/The, for consistency

File: app/request.go [hunk 9]
```diff
@@ -63,8 +88,93 @@ func NewRequestCommand(getRequestClientFn GetRequestClientFn) (CommandOut, error
...
+		case <-ticker.C:
+			status, err := c.getRequestStatus(ctx, requestID)
+			if err != nil {
+				return err
+			}
+			switch status.State {
+			case request.STATE_FULFILLED:
+				break loop
+			case request.STATE_FAILED:
+				fmt.Fprintf(writer, "operation failed \n")
```
Comment (feihuang) [inline app/request.go:144]: remove space after 'failed.' same below.

**Review by feihuang — APPROVED** [review]

**Pull request closed without merging**